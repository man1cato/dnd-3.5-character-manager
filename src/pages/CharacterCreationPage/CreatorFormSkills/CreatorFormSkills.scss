@import '../CharacterCreationPage.scss';


.CreatorFormSkills {
   display: flex;
   flex-direction: column;
   overflow: hidden;
}

.CreatorFormSkills__scroll-container {
   flex-grow: 1;
   display: flex;
   flex-direction: column;
   overflow: hidden;
}

.CreatorFormSkills__scroll-list {
   flex-grow: 1;
   overflow-y: auto;
   padding-top: $xs-size;
}

.CreatorFormSkills__grid {
   @extend .CreatorForm__grid;
   grid-template-columns: 2fr 1fr;
   grid-template-rows: auto;
   > button {
      padding-bottom: 0;
   }
}
.CreatorFormSkills__grid--header {
   @extend .CreatorFormSkills__grid;
   padding-right: 17px;
   padding-bottom: 0.5rem;
   border-bottom: 1px solid $dark-grey;
}

.CreatorFormSkills__counter {
   height: 4rem;
   button {
      height: 100%;
      padding: $xs-size !important; 
   }
   input {
      height: 100%;
      width: 100%;
	}
}