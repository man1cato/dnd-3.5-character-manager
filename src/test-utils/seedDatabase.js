import _ from 'lodash'
import profile from './fixtures/profile'
import database from '../firebase/firebase'


export const characterOne = {
   ..._.omit(profile, ['id'])
}

export const characterTwo = {
   name: 'Korg',
   gender: 'Male',
   race: 'recoDD3gAaQXU3A56',
   jobClass: 'recbxccm8pp3fTmQw',
   alignment: 'Lawful Evil',
   level: 5,
   iconUrl: 'images/orc.svg'
}

export const characterThree = {
   name: 'Rylia',
   gender: 'Female',
   race: 'recFEpQ9zuD9Yqu7l',
   jobClass: 'recQrfCKtksnnSsaZ',
   alignment: 'Chaotic Neutral',
   level: 3,
   iconUrl: 'images/halfling.svg'
}

export const userOne = {
   uid: 'userOne',
   characterIds: ['characterOne', 'characterTwo']
}

export const userTwo = {
   uid: 'userTwo',
   characterIds: ['characterThree']
}


export default async () => {
   await database.ref('users').remove()
   await database.ref(`users/${userOne.uid}/profiles/${userOne.characterIds[0]}`).set(characterOne)
   await database.ref(`users/${userOne.uid}/profiles/${userOne.characterIds[1]}`).set(characterTwo)
   await database.ref(`users/${userTwo.uid}/profiles/${userTwo.characterIds[0]}`).set(characterThree)
}