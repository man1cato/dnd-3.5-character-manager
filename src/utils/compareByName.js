export default (a,b) => {
    const aName = a.name.toUpperCase();
    const bName = b.name.toUpperCase();
    if (aName < bName) { return -1 };
    if (aName > bName) { return 1 };
    return 0;
}