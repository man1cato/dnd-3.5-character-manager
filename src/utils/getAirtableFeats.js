import dotenv from 'dotenv'
dotenv.config()

import axios from 'axios'
import _ from 'lodash'

const apiKey = process.env.AIRTABLE_API_KEY
const baseUrl = 'https://api.airtable.com/v0/appK7TZeddGqjGUDL'

export default async () => {
	try {
		let offset
		let feats = []

		do {
			const res = await axios.get(`${baseUrl}/Feats?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || '') )
			offset = res.data.offset
			const featGroup = res.data.records.map((feat) =>  ({
				id: feat.id,
				name: feat.fields.Name,
				type: feat.fields.Type.join(", ") || null,
				description: feat.fields.Description || null
			}));
			
			feats = feats.concat(featGroup)
		} while (!!offset)
		
		return _.orderBy(feats, ['name'], ['asc'])
	} catch (e) {
		console.error(e)
	}
}