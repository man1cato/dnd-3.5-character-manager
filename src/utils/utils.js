import _ from 'lodash'


export const convertTextToArray = (string) => _.compact(string.split(/\n/))

export const findItemById = (storeItemArray, itemId) => storeItemArray.find((item) => item.id === itemId)

export const apiObjectToArray = (obj) => _.orderBy(Object.keys(obj).map((id) => ({
	id,
	...obj[id]
})), ['name'], ['asc'])