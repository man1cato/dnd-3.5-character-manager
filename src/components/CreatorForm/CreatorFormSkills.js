import React, { useState, useEffect, Fragment } from 'react'
import _ from 'lodash'
import { Field, ErrorMessage } from 'formik'

import SkillModal from '../SkillModal'
import { apiObjectToArray } from '../../utils/utils'


const CreatorFormSkills = ({
	values,
	skills,
	setFieldValue
}) => { 
	const { skillSet, skillPoints } = values
	const [assignedSkillPoints, setAssignedSkillPoints] = useState(0)
	const [selected, setSelected] = useState(undefined)

	useEffect(() => {
		setAssignedSkillPoints(_.sumBy(skillSet, skill => skill.ranks))
	})

	useEffect(() => {
		setFieldValue('remainingSkillPoints', skillPoints - assignedSkillPoints)
	}, [assignedSkillPoints])
	

	return (
		<>
			<h3 className="row--center">Assign Skill Points</h3>
			<div className="row--center">{assignedSkillPoints}/{skillPoints}</div>
			<ErrorMessage className="row--center form-group--error" name="remainingSkillPoints" component="div" />
			
			<div className="form-grid--skills">
				<h5 className="grid__col1 form-grid__header">Skill</h5>
				<h5 className="grid__col3 form-grid__header">Ranks</h5>

				{apiObjectToArray(skills).map((skill, i) => {
					const fieldName = `skillSet[${i}].ranks`
					const ranks = skillSet[i].ranks
					return (
						<Fragment key={i}>
							<button
								className="grid__col1 button--link"	
								type="button"
								onClick={() => setSelected(skill)}
							>
								{skill.name}
							</button>
							
							<button 
								className="grid__col2 button" 
								type="button" 
								onClick={() => setFieldValue(fieldName, ranks > 0 ? ranks - 1 : 0)}
							>
								-
							</button>
							
							<Field 
								className="grid__col3 number-input"
								id={skill.id}
								name={fieldName}
								type="number"
							/>		

							<button 
								className="grid__col4 button" 
								type="button" 
								onClick={() => setFieldValue(fieldName, ranks + 1)}
							>
								+
							</button>
						</Fragment>
					)
				})}
			</div>

			
			<SkillModal
				selected={selected}
				handleCloseModal={() => setSelected(undefined)}
			/>

		</>
	)
}
export default CreatorFormSkills