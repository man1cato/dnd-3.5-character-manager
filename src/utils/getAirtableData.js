import axios from 'axios'
import _ from 'lodash'
import { convertTextToArray } from './utils' 
import dotenv from 'dotenv'
dotenv.config({ path: './config/dev.env' })

const apiKey = process.env.AIRTABLE_API_KEY
const baseUrl = 'https://api.airtable.com/v0/appK7TZeddGqjGUDL'


export const getFeats = async () => {
	try {
		let offset
		let feats = {}

		do {
			const res = await axios.get(`${baseUrl}/Feats?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || '') )
			offset = res.data.offset
			res.data.records.forEach((feat) => {
				feats[feat.id] = {
					name: feat.fields.Name,
					types: feat.fields.Types || null,
					description: feat.fields.Description || null,
					prerequisites: {
						level: feat.fields['Prereq. Level'] || null,
						class: feat.fields['Prereq. Class'] || null,
						feats: feat.fields['Prereq. Feats'] || null
					} || null
				}
			})			
		} while (!!offset)
		
		return feats
	} catch (e) {
		console.error(e)
	}
}


export const getItems = async () => {
	try {
		let offset
		let items = {}
		do {
			const res = await axios.get(`${baseUrl}/Items?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || ''))
			offset = res.data.offset
			res.data.records.forEach((item) => {
				items[item.id] = {
					name: item.fields.Name,
					value: item.fields["Value (gp)"] || '',
					weight: item.fields["Weight (lbs)"] || '',
					category: item.fields.Category || null,
					weaponType: item.fields["Weapon Type"] || null,
					encumbrance: item.fields.Encumbrance || null,
					damageType: item.fields["Damage Type"] ? item.fields["Damage Type"].join(" / ") : [],
					damageS: item.fields["Damage (S)"] || null,
					damageM: item.fields["Damage (M)"] || null,
					critical: item.fields.Critical || null,
					range: item.fields.Range || null,
					armorBonus: item.fields["Armor Bonus"] || null,
					maxDexBonus: item.fields["Max DEX Bonus"] || null
				}
			})
		} while (!!offset)
		
		return items
	} catch (e) {
		console.error(e)
	}
}


const getLevels = async (jobClassId) => {
	try {
		const levelRes = await axios.get(`${baseUrl}/Levels?filterByFormula={Class ID}="${jobClassId}"&api_key=${apiKey}`)
		
		const levelRecords = levelRes.data.records
		let levels = {}
		for (let i = 0; i < levelRecords.length; i++) {
			const levelRecord = levelRecords[i].fields 
			levels[levelRecord.Level] = {
				saves: {
					fortitude: levelRecord['Fortitude'],
					reflex: levelRecord['Reflex'],
					will: levelRecord['Will']
				},
				baseAttackBonuses: _.compact([ levelRecord['BAB 1'], levelRecord['BAB 2'], levelRecord['BAB 3'], levelRecord['BAB 4'] ]),
				xp: levelRecord['XP Required'],
				specialAbilities: !!levelRecord['Special Abilities'] ? levelRecord['Special Abilities'] : null,
				spellsPerDay:  _.compact([ 
					levelRecord['SPD 0'], levelRecord['SPD 1'], levelRecord['SPD 2'], levelRecord['SPD 3'], levelRecord['SPD 4'], 
					levelRecord['SPD 5'], levelRecord['SPD 6'], levelRecord['SPD 7'], levelRecord['SPD 8'], levelRecord['SPD 9'] 
				])
			}
		}
		return levels
	} catch (e) {
		console.error(e)
	}
}

export const getJobClasses = async () => {
	try {
		let offset
		let jobClasses = {}
	
		do {
			const classRes = await axios.get(`${baseUrl}/Classes?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || '') )
			offset = classRes.data.offset

			const jobClassGroup = classRes.data.records
			
			for (let i = 0; i < jobClassGroup.length; i++) {
				const jobClass = jobClassGroup[i] 
				jobClasses[jobClass.id] = {
					name: jobClass.fields.Name,
					hitDie: jobClass.fields['Hit Die'],
					proficiencies: jobClass.fields['Proficiencies'] || [],
					skills: jobClass.fields.Skills || [],
					levels: await getLevels(jobClass.id)
				}
			}
		} while (!!offset)
		
		return jobClasses
	} catch (e) {
		console.error(e)
	}
}


export const getRaces = async () => {
	try {
		let offset
		let races = {}
		do {
			const res = await axios.get(`${baseUrl}/Races?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || '') )
			offset = res.data.offset
			res.data.records.forEach((race) =>  {
				races[race.id] = {
					name: race.fields.Name,
					size: race.fields.Size,
					speed: race.fields['Speed (ft/rnd)'],
					racialBonuses: race.fields['Racial Bonuses'] ? convertTextToArray(race.fields['Racial Bonuses']) : null,
					defaultLanguages: race.fields['Languages (Default)'],
					bonusLanguages: race.fields['Languages (Bonus)'],
					abilityMods: {
						str: race.fields['STR Mod'],
						dex: race.fields['DEX Mod'],
						con: race.fields['CON Mod'],
						int: race.fields['INT Mod'],
						wis: race.fields['WIS Mod'],
						cha: race.fields['CHA Mod']
					},
					specialAbilities: race.fields['Special Abilities'] || [],
					favoredClass:  race.fields['Favored Class'] ? race.fields['Favored Class'][0] : null,
					iconUrl: race.fields['Icon URL']
				}
			})
			
		} while (!!offset)
		
		return races
	} catch (e) {
		console.error(e)
	}
}


export const getSkills = async () => {
   let offset
   let skills = {}
   do {
		const res = await axios.get(`${baseUrl}/Skills?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || '') )
		offset = res.data.offset
		res.data.records.forEach((skill) => {
			skills[skill.id] = {
				name: skill.fields.Name,
				keyAbility: skill.fields["Key Ability"] || null,
				description: skill.fields.Description || "",
				notes: skill.fields.Notes || ""
			}
		})
   } while (!!offset)
   
   return skills
}


export const getSpecialAbilities = async () => {
   let offset
   let specialAbilities = {}
   do {
		const res = await axios.get(`${baseUrl}/Special Abilities?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || ''))
		offset = res.data.offset
		res.data.records.forEach((specialAbility) => {
			specialAbilities[specialAbility.id] = {
				name: specialAbility.fields.Name,
				type: specialAbility.fields.Type ? specialAbility.fields.Type.join(", ") : null,
				description: specialAbility.fields.Description				
			}
		})
   } while (!!offset)
   
   return specialAbilities
}


export const getSpells =  async () => {
   let offset
   let spells = {}
   do {
		const res = await axios.get(`${baseUrl}/Spells?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || '')) 
		offset = res.data.offset
		res.data.records.forEach((spell) => {
			spells[spell.id] = {
				name: spell.fields.Name,
				description: spell.fields.Description || "",
				school: spell.fields.School || null,
				components: spell.fields.Components.join(", ") || null,
				castingTime: spell.fields["Casting Time"] || null,
				range: spell.fields.Range || null,
				effectArea: spell.fields["Target, Effect, Area"] || null,
				duration: spell.fields.Duration || null,
				savingThrow: spell.fields["Saving Throw"] || null,
				spellResistance: spell.fields["Spell Resistance"] || null
			}
		})
   } while (!!offset)

   return spells
}