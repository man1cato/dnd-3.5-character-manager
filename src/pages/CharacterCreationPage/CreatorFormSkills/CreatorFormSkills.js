import React, { useState, useEffect, Fragment } from 'react'
import _ from 'lodash'
import { ErrorMessage } from 'formik'

import Counter from '../../../components/Counter/Counter'
import SkillModal from '../../../components/Modals/SkillModal'
import { apiObjectToArray } from '../../../utils/utils'
import './CreatorFormSkills.scss'


const CreatorFormSkills = ({
	values,
	skills,
	setFieldValue
}) => { 
	const { skillSet, skillPoints } = values
	const [assignedSkillPoints, setAssignedSkillPoints] = useState(0)
	const [selected, setSelected] = useState(undefined)

	useEffect(() => {
		setAssignedSkillPoints(_.sumBy(skillSet, skill => skill.ranks))
	})

	useEffect(() => {
		setFieldValue('remainingSkillPoints', skillPoints - assignedSkillPoints)
	}, [assignedSkillPoints])
	
	const handleChange = (index, value) => {
		const ranks = value > 0 ? value : 0
		setFieldValue(`skillSet[${index}].ranks`, ranks)
	}

	return (
		<div className="container--body CreatorFormSkills">
			<h3 className="section__title">Assign Skill Points</h3>
			<div className="section__title">{assignedSkillPoints}/{skillPoints}</div>
			<ErrorMessage className="CreatorForm__error--center" name="remainingSkillPoints" component="div" />

			<div className="CreatorFormSkills__scroll-container">
				<div className="CreatorFormSkills__grid--header">
					<h5 className="grid__col1">Skill</h5>
					<h5>Ranks</h5>
				</div>
				<div className="CreatorFormSkills__scroll-list">
					<div className="CreatorFormSkills__grid">
						{apiObjectToArray(skills).map((skill, i) => {
							const ranks = skillSet[i].ranks
							return (
								<Fragment key={i}>
									<button
										className="grid__col1 button--link"	
										type="button"
										onClick={() => setSelected(skill)}
									>
										{skill.name}
									</button>

									<Counter
										className="CreatorFormSkills__counter"
										value={ranks}
										updateValue={value => handleChange(i, value)}
									/>							
								</Fragment>
							)
						})}
					</div>
				</div>
			</div>
			
			<SkillModal
				selected={selected}
				handleCloseModal={() => setSelected(undefined)}
			/>

		</div>
	)
}
export default CreatorFormSkills