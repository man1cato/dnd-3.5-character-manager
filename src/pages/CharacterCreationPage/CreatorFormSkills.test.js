import React from 'react'
import { shallow } from 'enzyme'
import _ from 'lodash'

import CreatorFormSkills from './CreatorFormSkills'
import { apiData } from '../../tests/utils/utils'
import { characterOne } from '../../tests/utils/seedDatabase'
import { apiObjectToArray } from '../../utils/utils'


const setFieldValue = jest.fn()
let wrapper, props
beforeAll(async () => {
   const api = await apiData()
   props = {
      values: { 
         skillSet: apiObjectToArray(api.skills).map(skill => ({ id: skill.id, ranks: 0 })),
         skillPoints: characterOne.skillPoints
      },
      skills: api.skills,
      setFieldValue
   }
   wrapper = shallow(<CreatorFormSkills {...props} />)
})


test('should render CreatorFormSkills correctly', () => {
	expect(wrapper).toMatchSnapshot()
})
