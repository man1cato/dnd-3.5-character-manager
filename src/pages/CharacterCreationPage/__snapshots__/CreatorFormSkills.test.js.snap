// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreatorFormSkills correctly 1`] = `
<div
  className="container--body CreatorFormSkills"
>
  <div>
    <h3
      className="section__title"
    >
      Assign Skill Points
    </h3>
    <div
      className="row--center"
    >
      0
      /
      12
    </div>
    <FormikConnect(ErrorMessageImpl)
      className="row--center CreatorForm__error"
      component="div"
      name="remainingSkillPoints"
    />
  </div>
  <div
    className="CreatorForm__scroll-container"
  >
    <div
      className="CreatorFormSkills__grid--header"
    >
      <h5
        className="grid__col1"
      >
        Skill
      </h5>
      <h5>
        Ranks
      </h5>
    </div>
    <div
      className="CreatorForm__scroll-list"
    >
      <div
        className="CreatorFormSkills__grid"
      >
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Appraise
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Balance
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Bluff
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Climb
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Concentration
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Control Shape
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Alchemy)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Armorsmithing)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Basketweaving)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Blacksmithing)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Bookbinding)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Bowmaking)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Carpentry)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Cartography)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Gemcutting)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Leatherworking)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Locksmithing)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Metalworking)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Poisonmaking)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Sculpting)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Shipmaking)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Stonemasonry)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Trapmaking)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Weaponsmithing)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Winemaking)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Craft (Woodmaking)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Decipher Script
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Diplomacy
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Disable Device
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Disguise
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Escape Artist
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Forgery
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Gather Information
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Handle Animal
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Heal
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Hide
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Intimidate
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Jump
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (Arcana)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (Arch. & Eng.)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (Dungeoneering)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (Geography)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (History)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (Local)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (Nature)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (Nobility & Royalty)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (Religion)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Knowledge (The Planes)
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Listen
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Literacy
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Move Silently
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Open Lock
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Perform
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Profession
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Ride
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Sense Motive
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Sleight of Hand
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Speak Language
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Spellcraft
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Spot
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Survival
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Swim
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Tumble
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Use Magic Device
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
        <button
          className="grid__col1 button--link"
          onClick={[Function]}
          type="button"
        >
          Use Rope
        </button>
        <Counter
          className="CreatorFormSkills__counter"
          updateValue={[Function]}
          value={0}
        />
      </div>
    </div>
  </div>
  <SkillModal
    handleCloseModal={[Function]}
  />
</div>
`;
